src/services/item/plugins/published/plugins/search/index.ts

```diff
@@ -17,12 +19,22 @@ export type SearchFields = {
 
 const plugin: FastifyPluginAsync = async (fastify) => {
   const searchService = fastify.search.service;
+  const actionService = fastify.actions.service;
 
   fastify.post(
     '/collections/search',
     { preHandler: fastify.attemptVerifyAuthentication, schema: search },
-    async ({ member, body }) => {
-      return searchService.search(member, buildRepositories(), body as MultiSearchParams);
+    async (request) => {
+      const { member, body } = request;
+      const repositories = buildRepositories();
+
+      // TODO: chnage type from graasp/sdk
+      const action = {
+        type: 'item-search',
+        extra: body as Partial<Action> & Pick<Action, 'type'>,
```

Comment:
You can precise the type like this if I'm not mistaken:
`fastify.post<{ Body: Partial<Action> & Pick<Action, 'type'> }>(`

Reply:
Also update the schema of the search endpoint accordingly.